src/platform/BUILD.gn

```diff
@@ -158,171 +158,187 @@ if (chip_device_platform != "none" && chip_device_platform != "external") {
-        "CHIP_DEVICE_LAYER_TARGET_MW320=1",
-        "CHIP_DEVICE_LAYER_TARGET=nxp/mw320",
-      ]
+      device_layer_target_define = "MW320"
+      defines += [ "CHIP_DEVICE_LAYER_TARGET=nxp/mw320" ]
     } else if (chip_device_platform == "zephyr") {
-      defines += [
-        "CHIP_DEVICE_LAYER_TARGET_ZEPHYR=1",
-        "CHIP_DEVICE_LAYER_TARGET=Zephyr",
-      ]
+      device_layer_target_define = "ZEPHYR"
+      defines += [ "CHIP_DEVICE_LAYER_TARGET=Zephyr" ]
     } else if (chip_device_platform == "beken") {
+      device_layer_target_define = "BEKEN"
       defines += [
-        "CHIP_DEVICE_LAYER_TARGET_BEKEN=1",
         "CHIP_DEVICE_LAYER_TARGET=Beken",
         "CHIP_DEVICE_CONFIG_ENABLE_WIFI=${chip_enable_wifi}",
       ]
     } else if (chip_device_platform == "mt793x") {
+      device_layer_target_define = "MT793X"
       defines += [
-        "CHIP_DEVICE_LAYER_TARGET_MT793X=1",
         "CHIP_DEVICE_LAYER_TARGET=mt793x",
         "CHIP_DEVICE_CONFIG_ENABLE_WIFI=${chip_enable_wifi}",
       ]
     } else if (chip_device_platform == "openiotsdk") {
-      defines += [
-        "CHIP_DEVICE_LAYER_TARGET_OPEN_IOT_SDK=1",
-        "CHIP_DEVICE_LAYER_TARGET=openiotsdk",
-      ]
+      device_layer_target_define = "OPEN_IOT_SDK"
+      defines += [ "CHIP_DEVICE_LAYER_TARGET=openiotsdk" ]
     } else if (chip_device_platform == "asr") {
-      defines += [
-        "CHIP_DEVICE_LAYER_TARGET_ASR=1",
-        "CHIP_DEVICE_LAYER_TARGET=ASR",
-      ]
+      device_layer_target_define = "ASR"
+      defines += [ "CHIP_DEVICE_LAYER_TARGET=ASR" ]
     } else if (chip_device_platform == "stm32") {
-      defines += [
-        "CHIP_DEVICE_LAYER_TARGET_STM32=1",
-        "CHIP_DEVICE_LAYER_TARGET=stm32",
-      ]
+      device_layer_target_define = "STM32"
+      defines += [ "CHIP_DEVICE_LAYER_TARGET=stm32" ]
+    } else {
+      device_layer_target_define = ""
     }
 
+    assert(device_layer_target_define != "",
+           "Unknown chip_device_platform value: '${chip_device_platform}'")
+
+    found_device_layer_target_define = false
+    possible_device_layer_target_defines = [
+      "CC13X2_26X2",
+      "CC13X4_26X4",
+      "CC32XX",
+      "DARWIN",
+      "EFR32",
+      "ESP32",
+      "PSOC6",
+      "LINUX",
+      "TIZEN",
+      "NRFCONNECT",
+      "QPG",
+      "K32W",
+      "TELINK",
+      "MBED",
+      "BL602",
+      "BL702",
+      "BL702L",
+      "FAKE",
+      "ANDROID",
+      "AMEBA",
+      "CYW30739",
+      "WEBOS",
+      "MW320",
+      "ZEPHYR",
+      "BEKEN",
+      "MT793X",
+      "OPEN_IOT_SDK",
+      "ASR",
+      "STM32",
+    ]
+    foreach(possible_device_layer_target_define,
+            possible_device_layer_target_defines) {
+      if (possible_device_layer_target_define == device_layer_target_define) {
+        defines += [
+          "CHIP_DEVICE_LAYER_TARGET_${possible_device_layer_target_define}=1",
+        ]
+      } else {
+        found_device_layer_target_define = true
```

Comment:
Shouldn't this be set in the `if` above rather than here?

Reply:
@Damian-Nordic Yes, it should! Good catch.  Fixed.